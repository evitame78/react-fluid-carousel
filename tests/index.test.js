import Carousel from "../src";
import React from "react";
import { render, cleanup } from "react-testing-library";

describe("Test the component", () => {
  beforeEach(() => {
    Element.prototype.getBoundingClientRect = jest.fn(() => {
      return {
        width: 40,
        height: 40,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0
      };
    });
  });
  afterEach(cleanup);
  test("Empty carousel", () => {
    expect(render(<Carousel />).container).toMatchSnapshot();
  });
  test("With some children", () => {
    const component = (
      <Carousel>
        <div key="1" style={{ width: "40px", height: "40px" }} />
        <div key="2" style={{ width: "40px", height: "40px" }} />
      </Carousel>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
