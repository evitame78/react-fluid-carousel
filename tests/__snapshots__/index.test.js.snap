// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component Empty carousel 1`] = `
<div>
  <div>
    Gimme some children
  </div>
</div>
`;

exports[`Test the component With some children 1`] = `
<div>
  <div
    style="width: 40px; height: 40px;"
  />
</div>
`;
